server/src-exec/Main.hs

```diff
@@ -129,8 +131,9 @@ main =  do
       putStrLn $ "server: running on port " ++ show port
       (app, cacheRef) <- mkWaiApp isoL loggerCtx pool httpManager
                          am corsCfg enableConsole
-      let warpSettings = Warp.setPort port Warp.defaultSettings
-                         -- Warp.setHost "*" Warp.defaultSettings
+
+      let hostS = DataString.fromString(host)
+      let warpSettings = Warp.setPort port $ Warp.setHost hostS Warp.defaultSettings
```

Comment:
You won't need to do the conversion as `ServerOptions` will have the value of correct type. You can get rid of `Data.String` import.